### PR TITLE
SK Disability amount

### LINKED ISSUE #370: SK Disability amount
Reference: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1sk/td1sk-23e.pdf

### PR DESCRIPTION
**Pull request by Yaohhhh**
Fixed #370

### CODE REVIEW

File: policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/sk_disability_amount/amount.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+description: Saskatchewan provides this tax-credit amount for the qualified disabled.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/sk_disability_amount/amount.yaml:1]:
```suggestion
description: Saskatchewan provides this amount for qualified disabled filers under the saskatchewan disability amount.
```

File: policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/sk_disability_amount/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+values:
+  2023-01-01: 10_405
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/sk_disability_amount/amount.yaml:3]: need 2022 value + 2022 form

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/sk_disability_amount/sk_disability_amount.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    entity = Person
+    label = "SK Disability Amount Tax Credit"
+    definition_period = YEAR
+    defined_for = ProvinceCode.SK
+
+    def formula(person, period, parameters):
+        p = parameters(
+            period
+        ).gov.provinces.sk.tax.income.credits.sk_disability_amount
+        eligibility = person("is_disable_certificate", period)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/sk_disability_amount/sk_disability_amount.py:15]:
```suggestion
        eligibility = person("is_disabled", period)
        head = person("is_head", period)
```

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/sk_disability_amount/sk_disability_amount.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        return where(eligibility == 1, p.amount, 0)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/sk_disability_amount/sk_disability_amount.py:16]:
```suggestion
        return where(eligibility, p.amount, 0)
```

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/sk_disability_amount/is_disable_certificate.py
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/sk_disability_amount/is_disable_certificate.py:1]: dont need this

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/sk_disability_amount/sk_disability_amount.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+from policyengine_canada.model_api import *
+
+
+class sk_disability_amount(Variable):
+    value_type = float
+    entity = Person
+    label = "SK Disability Amount Tax Credit"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/sk_disability_amount/sk_disability_amount.py:7]:
```suggestion
    label = "Saskatchewan Disability Amount Tax Credit"
```

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/sk_disability_amount/sk_disability_amount.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    definition_period = YEAR
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/sk_disability_amount/sk_disability_amount.py:8]: add: unit: CAD

File: changelog_entry.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Sk Disability amount tax credit.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Saskatchewan Disability amount.
```

File: policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/sk_disability_amount/amount.yaml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  unit: currency-CAD
+  label: Saskatchewan disability tax credit amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/sk_disability_amount/amount.yaml:6]:
```suggestion
  label: Saskatchewan disability amount
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/disability_amount/amount.yaml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+description: Saskatchewan provides this amount for qualified disabled filers under the disability amount.
+values:
+  2022-01-01: 9_789
+  2023-01-01: 10_405
+metadata:
+  unit: currency-CAD
+  label: Saskatchewan disability amount
+  reference:
+  - title: 2023 Saskatchewan TD1SK Personal Tax Credits Return
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/disability_amount/amount.yaml:9]: and elsewhere
```suggestion
  - title: 2023 Saskatchewan TD1SK Personal Tax Credits Return, line 5
```

File: policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/disability_amount/amount.yaml [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1sk/td1sk-23e.pdf
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/disability_amount/amount.yaml:10]: and elsewhere
```suggestion
    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1sk/td1sk-23e.pdf#page=1
```

File: policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/disability_amount/amount.yaml [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  - title: Saskatchewan I2-01 The Income Tax Act, 2000, Page 25, Mental or physical impairment credit (23), (1)
+    href: https://pubsaskdev.blob.core.windows.net/pubsask-prod/806/I2-01.pdf
+  - title: 2022 Saskatchewan TD1SK Personal Tax Credits Return
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1sk/td1sk-22e.pdf
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/disability_amount/amount.yaml:14]: Cite legal code or leave a comment saying it's unavailable

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py [hunk 12]
```diff
@@ -0,0 +1,19 @@
+from policyengine_canada.model_api import *
+
+
+class sk_disability_amount(Variable):
+    value_type = float
+    entity = Person
+    unit: CAD
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py:7]:
```suggestion
    unit = CAD
```

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+    label = "Saskatchewan Disability Amount Tax Credit"
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py:8]:
```suggestion
    label = "Saskatchewan disability amount tax credit"
```

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+    definition_period = YEAR
+    defined_for = ProvinceCode.SK
+
+    def formula(person, period, parameters):
+        p = parameters(
+            period
+        ).gov.provinces.sk.tax.income.credits.disability_amount
+        disabled = person("is_disabled", period)
+        is_head = person("is_head", period)
+        eligbile = is_head & disabled
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py:18]:
```suggestion
        eligible = is_head & disabled
        return eligible * p.amount
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+    label = "Saskatchewan disability amount tax credit"
+    definition_period = YEAR
+    defined_for = ProvinceCode.SK
+
+    def formula(person, period, parameters):
+        p = parameters(
+            period
+        ).gov.provinces.sk.tax.income.credits.disability_amount
+        disabled = person("is_disabled", period)
+        is_head = person("is_head", period)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py:17]:
```suggestion
```

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+        eligbile = is_head & disabled
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py:18]:
```suggestion
```

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+        return eligible * p.amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py:19]:
```suggestion
        return disabled * p.amount
```

File: policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/disability_amount/amount.yaml [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+  2022-01-01: 9_789
+  2023-01-01: 10_405
+metadata:
+  unit: currency-CAD
+  label: Saskatchewan disability amount
+  reference:
+  - title: 2023 Saskatchewan TD1SK Personal Tax Credits Return, line 5
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1sk/td1sk-23e.pdf#page=1
+  - title: Saskatchewan I2-01 The Income Tax Act, 2000, Page 25, Mental or physical impairment credit (23), (1)
+    href: https://pubsaskdev.blob.core.windows.net/pubsask-prod/806/I2-01.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/sk/tax/income/credits/disability_amount/amount.yaml:12]: add page number to pdf here

File: policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+    definition_period = YEAR
+    defined_for = ProvinceCode.SK
+
+    def formula(person, period, parameters):
+        p = parameters(
+            period
+        ).gov.provinces.sk.tax.income.credits.disability_amount
+        disabled = person("is_disabled", period)
+        is_head = person("is_head", period)
+        eligible = is_head & disabled
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/sk/tax/income/credits/disability_amount/sk_disability_amount.py:18]:
```suggestion
```

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**